summarize/Readme.md

```diff
@@ -2,17 +2,48 @@
 
 Summarize is a tool to produce a human readable summary of `measureme` profiling data.
 
-## Example
+## Installing summarize
+
+In order to use this tool you will first want to build it. To do this follow the steps
+below:
 
 ```bash
-$ git clone https://github.com/rust-lang/regex.git
+$ git clone https://github.com/rust-lang/measureme.git
+$ cd measureme/summarize
+$ cargo +nightly build --release
+```
+
+This will compile the `summarize` binary which we will need later. It will be located in
+the `measureme/target/release` folder (ie. not in the `summarize` folder).
+
+## Profiling the nightly compiler
+
+To profile the nightly compiler first ensure that you have a recent nightly compiler by
+typing `rustup update nightly`. If your compiler version is older than `2019-04-13` the
+profiling feature has not yet been added.
+
+Profiling the compiler is done by passing the flag `-Z self-profile` to `rustc`. Note that
+`-Z` flags are unstable, so you must use the nightly compiler. As an example we will
+profile the [regex][regex-crate] crate.
 
+[regex-crate]: https://github.com/rust-lang/regex
+
+```bash
+$ git clone https://github.com/rust-lang/regex.git
 $ cd regex
+$ cargo +nightly rustc -- -Z self-profile
+```
 
-$ cargo rustc -- -Z self-profile
+The commands above will run `rustc` with the flag that enables profiling. You should now
+have three files in your directory named `pid-{pid}.events`, `pid-{pid}.string_data` and
```

Comment:
We should change the naming scheme here to something like `rustc-profile-{pid}.{extension}`. 

Overall, I agree with the sentiment that the user shouldn't have to deal with 3 files here. They also shouldn't have to pass the `-Z self-profile` flag to `cargo`/`rustc` manually. We should try to collect workflows in a structured manner somewhere and then design the tooling interface from that.

Reply:
We had a few conversations about this in zulip, I think someone mentioned `summarize diff +nightly +mytoolchain` as a stretch goal for the syntax.  
However until we actually implement that, I think the docs should reflect the current state of things. We can always update them when we change the file format.

Reply:
>  I think someone mentioned summarize diff +nightly +mytoolchain as a stretch goal for the syntax.

Let's say it was a proposal `:)` No design has been agreed upon (or even discussed much).

>  until we actually implement that, I think the docs should reflect the current state of things.

Yes, I agree.